rust/kernel/seq_file.rs

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Trait for defining `seq_file`s.
+//!
+//! This module allows Rust devices to implement [`struct seq_operations`] and
+//! and create a file under `/proc` based on that implementation.
+//!
+//! C header: [`include/linux/seq_file.h`](../../../include/linux/seq_file.h)
+//!
+//! Reference: <https://www.kernel.org/doc/html/latest/filesystems/seq_file.html>
+
+// Currently this module is only usable through proc_fs.
+#![cfg_attr(not(CONFIG_PROC_FS), allow(dead_code))]
```

Comment:
This only adds the `dead_code` attribute when `CONFIG_PROC_FS` is not set. Since (currently) the only way to use the code here is through `proc_fs` we get some dead code warnings when `proc_fs` isn't enabled.

I changed it to put `#![cfg(CONFIG_PROC_FS)]` on the whole module. As we add other uses of `seq_file` we can expand that constraint.